@xen-orchestra/lite/src/components/Tree.tsx

```diff
@@ -114,14 +116,18 @@ const renderItem = ({ children, id, label, to, tooltip }: ItemType) => {
 
 const Tree = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
   {
-    initialState: ({ defaultSelectedNodes }) => ({
-      selectedNodes: defaultSelectedNodes === undefined ? [] : defaultSelectedNodes,
+    initialState: () => ({
+      _selectedNodes: undefined,
     }),
     effects: {
       setSelectedNodeIds: function (event, nodeIds) {
-        this.state.selectedNodes = nodeIds
+        this.state._selectedNodes = nodeIds
       },
     },
+    computed: {
+      selectedNodes: ({ _selectedNodes }, { defaultSelectedNodes }) =>
```

Comment:
We took the "default value" direction at first, but if it's easier, we can switch to a fully controlled component, if you want.